boards/components/src/button.rs

```diff
@@ -20,19 +21,40 @@ use kernel::static_init;
 
 #[macro_export]
 macro_rules! button_component_helper {
+    ($(($P:expr, $M:expr, $F:expr)),+ ) => {{
+        use kernel::static_init;
+        use kernel::count_expressions;
+        use kernel::hil::gpio::InterruptValueWrapper;
+        const NUM_BUTTONS: usize = count_expressions!($($P),+);
+
+        static_init!(
+            [(&'static dyn kernel::hil::gpio::InterruptValuePin, capsules::button::GpioMode, Option<kernel::hil::gpio::FloatingState>); NUM_BUTTONS],
+            [
+                $(
+                    (static_init!(InterruptValueWrapper, InterruptValueWrapper::new($P))
+                    .finalize(),
+                    $M,
+                    Some($F)
+                    ),
+                )*
+            ]
+        )
+    };};
+
     ($(($P:expr, $M:expr)),+ ) => {{
```

Comment:
I think it's fine to force users to specify how they want their GPIO internal pull up/down set, and just remove this case.

Reply:
Ok.
I wasn't sure and also couldn't decide which state to put for all the boards. Instinctively I would say that if the button is active low, we use pull-up and if it's active high we use pull-down. But it's also board specific because if you already have external pull-{up,down} resistors you probably want to disable the internal ones.

Reply:
> But it's also board specific because if you already have external pull-{up,down} resistors you probably want to disable the internal ones.

Yeah, I actually like the idea of forcing board authors to consider it.

Reply:
Done.